pages/common/xargs.md

```diff
@@ -1,19 +1,20 @@
 # xargs
 
-> Execute a command with piped arguments.
+> Execute a command with piped arguments coming from another command, a file, etc.
+> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
 
-- Main use:
+- Main usage pattern:
 
-`{{arguments}} | xargs {{command}}`
+`{{arguments_source}} | xargs {{command}}`
 
-- Specific example: delete all files that start with 'M':
+- Delete all files with a `.backup` extension:
 
-`find . -name 'M*' | xargs rm`
+`{{find . -name '*.backup'}} | xargs {{rm -v}}`
 
-- Handle whitespace in arguments:
+- Use the NUL (`\0`) character as separator (useful to split the input on newlines but not on spaces):
```

Comment:
I'd rather show a generic way this can work with any command, than rely on `find`'s custom behavior. I would keep the current example intact, i.e.

    `{{arguments_source}} | tr '\n' '\0' | xargs -0 {{command}}`

As for the wording... how about "Convert newlines in the input into NUL (\0) characters, and split on those only (useful if the input to xargs contains spaces)"? This, coupled with the main description, should provide both the "what" and the "how".